ClosedXML/Excel/Cells/XLCell.cs

```diff
@@ -1288,7 +1317,25 @@ private IEnumerable<XLCell> GetAffectingCells()
         /// </summary>
         private long NeedsRecalculationEvaluatedAtVersion { get; set; }
 
-        public object CachedValue { get; private set; }
+        private Object cachedValue;
+
+        public Object CachedValue
+        {
+            get
+            {
+                if (!HasFormula && cachedValue == null)
+                    cachedValue = Value;
+
+                return cachedValue;
+            }
+            private set
+            {
+                if (value != null && !HasFormula)
+                    throw new InvalidOperationException("Cached values can be set only for cells with formulas");
```

Comment:
Maybe add a test for this?

Reply:
Well, currently that exception cannot occur. It's a private setter, and where it's used it's always either `value == null` or `HasFormula == true`. Included this more for future-proofing the code. I don't really want to change the setter to `internal` just for the sake of testing an exception that doesn't currently occur. You ok with that?

Reply:
OK